### PR TITLE
Graphics for NXP Release 5.15.32-2.0.0

### PR DESCRIPTION
**Pull request by thochstein**

### CODE REVIEW

File: recipes-graphics/wayland/weston_10.0.0.imx.bb [hunk 1]
```diff
@@ -3,37 +3,34 @@
...
 UPSTREAM_CHECK_URI = "https://wayland.freedesktop.org/releases.html"
 
-inherit meson pkgconfig useradd features_check
+inherit meson pkgconfig useradd
+
 # depends on virtual/egl
-# weston-init requires pam enabled if started via systemd
-REQUIRED_DISTRO_FEATURES = "opengl ${@oe.utils.conditional('VIRTUAL-RUNTIME_init_manager', 'systemd', 'pam', '', d)}"
+#
+require ${THISDIR}/required-distro-features.inc
```
Comment (otavio) [inline recipes-graphics/wayland/weston_10.0.0.imx.bb:33]: please inline the content.

**Review by otavio — CHANGES_REQUESTED** [review]

File: conf/machine/include/imx-base.inc [hunk 2]
```diff
@@ -463,6 +463,7 @@ PREFERRED_PROVIDER_virtual/egl:imxgpu        ?= "imx-gpu-viv"
 PREFERRED_PROVIDER_virtual/libgl:imxgpu3d    ?= "imx-gpu-viv"
 PREFERRED_PROVIDER_virtual/libgles1:imxgpu3d ?= "imx-gpu-viv"
 PREFERRED_PROVIDER_virtual/libgles2:imxgpu3d ?= "imx-gpu-viv"
+PREFERRED_PROVIDER_virtual/libgles3:imxgpu3d ?= "imx-gpu-viv"
```
Comment (otavio) [inline conf/machine/include/imx-base.inc:466]: please add the changelog in commitlog.

File: recipes-graphics/wayland/weston_10.0.0.imx.bb (hunk 1 above)
Comment (thochstein) [reply recipes-graphics/wayland/weston_10.0.0.imx.bb:33]: This is how the OE-Core recipe does it.

Comment (otavio) [reply recipes-graphics/wayland/weston_10.0.0.imx.bb:33]: Arrg... ok.

File: recipes-kernel/kernel-modules/kernel-module-isp-vvcam_4.2.2.18.0.bb [hunk 3]
```diff
@@ -4,14 +4,10 @@ DESCRIPTION = "Kernel loadable module for ISP"
 LICENSE = "GPL-2.0-only"
 LIC_FILES_CHKSUM = "file://${WORKDIR}/git/vvcam/LICENSE;md5=64381a6ea83b48c39fe524c85f65fb44"
 
-SRCBRANCH = "lf-5.10.y_2.2.0"
-ISP_KERNEL_SRC ?= "git://source.codeaurora.org/external/imx/isp-vvcam.git;protocol=https;branch=master"
-
-SRC_URI = " \
-    ${ISP_KERNEL_SRC};branch=${SRCBRANCH} \
-    file://0001-dwe_isr.c-fix-error-found-by-gcc12.patch;patchdir=${WORKDIR}/git \
```
Comment (angolini) [inline recipes-kernel/kernel-modules/kernel-module-isp-vvcam_4.2.2.18.0.bb:12]: can we delete this patch file?

File: recipes-graphics/wayland/weston_10.0.0.imx.bb [hunk 4]
```diff
@@ -3,37 +3,34 @@
...
 SRC_URI = "https://wayland.freedesktop.org/releases/${BPN}-${PV}.tar.xz \
            file://weston.png \
            file://weston.desktop \
            file://xwayland.weston-start \
            file://systemd-notify.weston-start \
-           file://0001-weston-launch-Provide-a-default-version-that-doesn-t.patch \
-           file://0001-tests-include-fcntl.h-for-open-O_RDWR-O_CLOEXEC-and-.patch \
-           file://0001-meson.build-fix-incorrect-header.patch \
-           file://0001-libweston-backend-drm-Re-order-gbm-destruction-at-DR.patch \
-           file://0001-g2d-renderer-Add-vsync-to-cloned-displays.patch \
```
Comment (angolini) [inline recipes-graphics/wayland/weston_10.0.0.imx.bb:24]: can we remove all/some of those patch files?

**Review by angolini — APPROVED** [review]
I could not understand why there are so many changes in the line order with (what I think is) no change in the content, but I assume there is no reason.
For me the PR is good, only minor comments

**Comment by thochstein** [comment]
> I could not understand why there are so many changes in the line order with (what I think is) no change in the content, but I assume there is no reason.

Correct, no reason other than I think it greatly improves readability in a block to have the main variable in a block at the top, like SRC_URI, with any constituent and/or related variables listed underneath.

File: recipes-kernel/kernel-modules/kernel-module-isp-vvcam_4.2.2.18.0.bb (hunk 3 above)
Comment (thochstein) [reply recipes-kernel/kernel-modules/kernel-module-isp-vvcam_4.2.2.18.0.bb:12]: Yes, thanks for noticing.

File: recipes-graphics/wayland/weston_10.0.0.imx.bb (hunk 4 above)
Comment (thochstein) [reply recipes-graphics/wayland/weston_10.0.0.imx.bb:24]: Yes, thanks.

File: conf/machine/include/imx-base.inc (hunk 2 above)
Comment (thochstein) [reply conf/machine/include/imx-base.inc:466]: Will do.

**Review by otavio — APPROVED** [review]

**Pull request merged**